cpp/src/arrow/compute/kernels/scalar_nested.cc

```diff
@@ -17,6 +17,7 @@
 
 // Vector kernels involving nested types
 
+#include <iostream>
```

Comment:
Added for debugging, will remove later. Sorry for all the print statements that are littered throughout.

Reply:
Note you can use `arrow/util/logging.h` and `ARROW_LOG(WARNING) << ...;` (that way you don't have to deal with std::endl/the header might already be included, also, you'll get line numbers)